qiskit_nature/second_q/operators/sparse_label_op.py

```diff
@@ -32,6 +32,16 @@
 from .polynomial_tensor import PolynomialTensor
 
 
+_TCoeff = Union[complex, ParameterExpression]
+
+
+def _to_number(a: SupportsComplex | ParameterExpression) -> complex | float:
+    if isinstance(a, ParameterExpression):
+        sympified = a.sympify()
+        return complex(sympified) if sympified.is_Number else np.nan
+    return complex(a)
```

Comment:
I am not happy about specifically casting everything to `complex` here. Is that truely necessary? We should be able to just do:
```suggestion
    return a
```

Reply:
Done.